src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -67,6 +66,32 @@ public static Entity getEntityFromDatastore(String entityName, String entityProp
     return entity;
   }
 
+     /**
+    * This function takes in a list of cookies and matches a sessionID cookie
+    * with a specific name/value pair to a user entity with the same name/value
+    * pair as one of its properties. This user entity is then returned. 
+    * 
+    * <p>
+    * 
+    * @param  cookies    an array of cookies (usually all cookies on front end)
+    * @param  datastore  the database where entities are stored
+    * @return            a single entity that has the cookie name/value pair as a property
+    */
+ 
+  public static Entity getUserFromCookie(Cookie cookies[], DatastoreService datastore){
```

Comment:
This is a great concept. It's exactly how you should be determining who the current user is. Good.